### PR TITLE
feat(xlsx): initial support for xlsx data format, identity dsio

### PR DESCRIPTION
**Pull request by b5**
in a world where we have more time I'd break these commits up. This adds initial support for the xlsx data format as readers/writers.

Also added: initial support for "identity" dsio read/write for data that's already been parsed. We're going to need this later on for working with inlined dataset bodies. currently not in use, and shouldn't be used until we have tests.

The one change this does require is adding a `Close` method on dsio.EntryReader, which to me should probably be there anyway.

also fixed an issue with dsutil.FormFileDataset that wasn't properly assigning form values to the returned dataset.

### CODE REVIEW

File: data_format.go [hunk 1]
```diff
@@ -30,9 +30,9 @@ const (
 	// XMLDataFormat specifies eXtensible Markup Language-formatted data
 	// currently not supported.
 	XMLDataFormat
-	// XLSDataFormat specifies microsoft excel formatted data
+	// XLSXDataFormat specifies microsoft excel formatted data
 	// currently not supported.
```
Comment (dustmop) [inline data_format.go]: Remove "currently not supported". I understand this is only partial support, but I think it's enough that it moves beyond "not supported".

File: data_format_config.go [hunk 2]
```diff
@@ -141,3 +143,44 @@ func (o *JSONOptions) Map() map[string]interface{} {
...
+}
+
+// Format announces the XLSX data format for the FormatConfig interface
+func (*XLSXOptions) Format() DataFormat {
+	return XLSXDataFormat
+}
+
+// Map structures XLSXOptions as a map of string keys to values
+func (o *XLSXOptions) Map() map[string]interface{} {
+	if o == nil {
```
Comment (dustmop) [inline data_format_config.go]: Do we need to check nil here? In the rest of codebase, we're assuming that interface types won't contains a nil pointer value, for example see lib/datasets.go. Seems like an anti-pattern to support it here, and the code will just crash a few lines below where we try to resolve o.SheetName.

File: dsio/xslx.go [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+	}
+	if rdr.sheetName == "" {
+		rdr.sheetName = "Sheet1"
+	}
+
+	if rdr.err == nil {
+		rdr.r, rdr.err = rdr.file.Rows(rdr.sheetName)
+	}
+
+	return rdr, nil
```
Comment (dustmop) [inline dsio/xslx.go]: I'm unsure, but should this be `return rdr, rdr.err`? Otherwise the error from `Rows` is being ignored.

File: dsio/xslx.go [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+// ColIndexToLetters is used to convert a zero based, numeric column
+// indentifier into a character code.
+func ColIndexToLetters(colRef int) string {
+	parts := intToBase26(colRef)
+	return formatColumnName(smooshBase26Slice(parts))
+}
+
+// Get the largestDenominator that is a multiple of a basedDenominator
+// and fits at least once into a given numerator.
+func getLargestDenominator(numerator, multiple, baseDenominator, power int) (int, int) {
```
Comment (dustmop) [inline dsio/xslx.go]: I realize this function is coming from another source, but I find it pretty confusing, and a bit over complicated. How about:

```
// largestLesserExponent returns the largest exponent of base that does not exceed num
// equivalent to pow(floor(log(num, base)), base)
func largestLesserExponent(num, base int) int {
  prev := 1
  exp := base
  while num >= exp {
    prev = exp
    exp = exp * base
  }
  return prev
}
```

Comment (b5) [reply dsio/xslx.go]: this is delightful.

**Pull request merged**